app/scripts/contentscript.js

```diff
@@ -68,94 +69,174 @@ function injectScript(content) {
-  const pageStream = new WindowPostMessageStream({
+const setupPageStreams = () => {
+  console.log('setupPageStreams called');
+
+  pageStream = new WindowPostMessageStream({
     name: CONTENT_SCRIPT,
     target: INPAGE,
   });
-  const extensionPort = browser.runtime.connect({ name: CONTENT_SCRIPT });
-  const extensionStream = new PortStream(extensionPort);
 
-  // create and connect channel muxers
-  // so we can handle the channels individually
-  const pageMux = new ObjectMultiplex();
+  pageMux = new ObjectMultiplex();
   pageMux.setMaxListeners(25);
-  const extensionMux = new ObjectMultiplex();
+
+  extensionMux = new ObjectMultiplex();
   extensionMux.setMaxListeners(25);
   extensionMux.ignoreStream(LEGACY_PUBLIC_CONFIG); // TODO:LegacyProvider: Delete
 
   pump(pageMux, pageStream, pageMux, (err) =>
     logStreamDisconnectWarning('MetaMask Inpage Multiplex', err),
   );
+
   pump(extensionMux, extensionStream, extensionMux, (err) => {
     logStreamDisconnectWarning('MetaMask Background Multiplex', err);
     notifyInpageOfStreamFailure();
   });
 
-  // forward communication across inpage-background for these channels only
+  /** forward communication across inpage-background for these channels only */
   forwardTrafficBetweenMuxes(PROVIDER, pageMux, extensionMux);
 
-  // connect "phishing" channel to warning system
-  const phishingStream = extensionMux.createStream('phishing');
+  /** connect "phishing" channel to warning system */
+  phishingStream = extensionMux.createStream('phishing');
   phishingStream.once('data', redirectToPhishingWarning);
+};
+
+/**
+ * Destroying the destination streams of the pump (should) destroy the streams before it
+ *
+ * @todo
+ * - Remove text (should) above after confirming this works
+ * - Answer questions:
+ *   1. is it necessary to manually set allowHalfOpen to false?
+ *      "If false then the stream will automatically end the writable side when the readable side ends"
+ *   2 is it necessary to removeAllListeners if we're calling destroy?
+ *
+ * - is there an unpump method similar to unpipe?
+ * @see {@link https://nodejs.org/dist/v18.0.0/docs/api/stream.html#duplex-and-transform-streams}
+ * @see {@link https://github.com/mafintosh/pump#readme}
+ */
+const destroyPageStreams = () => {
+  extensionMux.allowHalfOpen = false;
+  extensionMux.removeAllListeners();
+  extensionMux.destroy();
+
+  pageMux.allowHalfOpen = false;
+  pageMux.removeAllListeners();
+  pageMux.destroy();
+
+  legacyPageMux.allowHalfOpen = false;
+  legacyPageMux.removeAllListeners();
+  legacyPageMux.destroy();
+
+  legacyExtensionMux.allowHalfOpen = false;
+  legacyExtensionMux.removeAllListeners();
+  legacyExtensionMux.destroy();
+
+  pageStream.allowHalfOpen = false;
+  pageStream.removeAllListeners();
+  pageStream.destroy();
+};
+
+/**
+ * Resets the extension stream with new streams and attach event listeners to the extension port.
+ */
+const resetSteamAndListeners = () => {
```

Comment:
It's not causing any issues but `resetSteamAndListeners` is missing a `r`; should be `resetStreamAndListeners`